src/wallet.js

```diff
@@ -876,6 +876,51 @@ const wallet = {
     return data;
   },
 
+  /**
+   * Filter an utxo based on the specified options
+   *
+   * @param {object} output Transaction output to be filtered/validated
+   * @param {object} utxoDetails utxos and meta information
+   * @param {object} options Consolidation options shared with getUtxos and consolidateUtxos (https://github.com/HathorNetwork/hathor-wallet-headless/issues/44)
+   *
+   * @return {object} { ..[rule]: boolen.. } object with each validation rule as property
+   */
+  filterUtxos(output, utxoDetails, options) {
+    const filterOptions = Object.assign({
+      max_utxos: MAX_INPUTS,
+      token: HATHOR_TOKEN_CONFIG.uid,
+      filter_address: null,
+      amount_smaller_than: Infinity,
+      amount_bigger_than: 0,
+      maximum_amount: Infinity
+    }, options);
+
+    // Filter by address, if options.filter_address is specified
+    const is_address_valid = filterOptions.filter_address === null || filterOptions.filter_address === output.decoded.address;
+    // Filter by maximum_amount (sum of utxos amounts), if options.maximum_amount is specified
+    const is_max_amount_valid = filterOptions.maximum_amount >= utxoDetails.total_amount_available;
```

Comment:
Here you must validate if the `filterOptions.maximum_amount >= utxoDetails.total_amount_available + output.value` because you must validate if this utxo can be added.